awsdk failed to compile on alpine

```bash
HEAD is now at ac02e17 Define STATIC_CRT (defaults to OFF) so that users can specify static/dynamic run-time library.
In file included from /TileDB/build/externals/src/ep_awssdk-build/.deps/build/src/AwsCCommon/tests/array_list_test.c:19:
/TileDB/build/externals/src/ep_awssdk-build/.deps/build/src/AwsCCommon/include/aws/testing/aws_test_harness.h:448:14: fatal error: execinfo.h: No such file or directory
 ```

Here is the relevant docker file
```
FROM alpine as builder
#prepare build env
RUN apk add g++ boost-dev boost-static git lapack-dev zlib-dev cmake make bash blas-dev openssl-dev
#build tiledb
RUN git clone https://github.com/TileDB-Inc/TileDB --depth=1 --branch=dev --single-branch
RUN cd TileDB && mkdir build && cd build && ../bootstrap --enable-s3 && make -j4 && make install-tiledb
```

making progress, after `apk add libexecinfo-dev `, it passed compiling but failed at linking 
```bash
make[3]: Leaving directory '/TileDB/build/externals/src/ep_awssdk-build/.deps'
-- Configuring incomplete, errors occurred!

-- stderr output is:
...skipping to end...
): undefined reference to `backtrace'
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: byte_buf_test.c:(.text+0xe7): undefined reference to `backtrace_symbols'
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: CMakeFiles/aws-c-common-tests.dir/byte_order_test.c.o: in function `s_print_stack_trace':
```

--- a/HISTORY.md
+++ b/HISTORY.md
@@ -13,6 +13,7 @@
 ## Improvements
 
 * Source built curl only need HTTP support [#1712](https://github.com/TileDB-Inc/TileDB/pull/1712)
+* AWS SDK version bumped to 1.8.6 [#1718](https://github.com/TileDB-Inc/TileDB/pull/1718)
 
 ## Deprecations
 

--- a/cmake/Modules/FindAWSSDK_EP.cmake
+++ b/cmake/Modules/FindAWSSDK_EP.cmake
@@ -96,8 +96,8 @@ if (NOT AWSSDK_FOUND)
 
     ExternalProject_Add(ep_awssdk
       PREFIX "externals"
-      URL "https://github.com/aws/aws-sdk-cpp/archive/1.7.108.zip"
-      URL_HASH SHA1=d07bae3fe924008b3117936963456dd314787abf
+      URL "https://github.com/aws/aws-sdk-cpp/archive/1.8.6.zip"
+      URL_HASH SHA1=5f4f58adabe2c7a241d49cb3ab2c96962fed1466
       CMAKE_ARGS
         -DCMAKE_BUILD_TYPE=${AWS_CMAKE_BUILD_TYPE}
         -DENABLE_TESTING=OFF

